tests/queries/0_stateless/00597_push_down_predicate_long.reference

```diff
@@ -68,67 +74,74 @@ FROM
     SELECT
         1 AS a,
         1 AS b
+    FROM system.one
 )
 ANY FULL OUTER JOIN
 (
     SELECT 1 AS a
+    FROM system.one
 ) USING (a)
 WHERE b = 0
 -------Need push down-------
 SELECT toString(value) AS value
 FROM
 (
     SELECT 1 AS value
+    FROM system.one
 )
+WHERE value = \'1\'
 1
-SELECT id
+SELECT *
 FROM
 (
-    SELECT 1 AS id
+    (
+        SELECT 1 AS id
+        FROM system.one
+    )
     UNION ALL
-    SELECT 2 AS `--predicate_optimizer_0`
-    WHERE 0
+    (
+        SELECT 2
+        FROM system.one
+    )
 )
 WHERE id = 1
 1
-SELECT id
+SELECT *
 FROM
 (
-    SELECT arrayJoin([1, 2, 3]) AS id
-    WHERE id = 1
+    SELECT arrayJoin(_CAST([1, 2, 3], \'Array(UInt8)\')) AS id
+    FROM system.one
 )
 WHERE id = 1
 1
 SELECT id
 FROM
 (
-    SELECT arrayJoin([1, 2, 3]) AS id
-    WHERE id = 1
+    SELECT arrayJoin(_CAST([1, 2, 3], \'Array(UInt8)\')) AS id
+    FROM system.one
 )
 WHERE id = 1
 1
-SELECT
-    id,
-    subquery
+SELECT *
```

Comment:
That looks suspicious. `EXPLAIN SYNTAX` should produce query without matchers.

Reply:
It's because we do not perform query tree analysys. In old analyzer we replace marchers on syntax level, in new one only if passes are run. I set running passes disabled by default because as you mention we want to keep `EXPLAIN SYNTAX` more user friendly (comparing to developer friendly `EXPLAIN QUERY TREE`), so it just performs syntax analysis without semantic analysis which is done in passes

Reply:
As I understand, the purpose of `EXPLAIN SYNTAX` is to see what the query looks like after semantic analysis and AST-level optimizations.